Library/Homebrew/formula.rb

```diff
@@ -3102,6 +3110,56 @@ def license(args = nil)
       end
     end
 
+    # @!attribute [w] allow_network_access
+    # The phases for which network access is allowed. By default, network
+    # access is allowed for all phases. Valid phases are `:build`, `:test`,
+    # and `:postinstall`. When no argument is passed, network access will be
+    # allowed for all phases.
+    # <pre>allow_network_access</pre>
+    # <pre>allow_network_access :build</pre>
+    # <pre>allow_network_access [:build, :test]</pre>
+    def allow_network_access(phases = [])
+      phases_array = Array(phases)
+      if phases_array.empty?
+        @network_access_allowed = @supported_phases.zip([true] * @supported_phases.length).to_h
```

Comment:
Might be worth splitting this `zip`/`length`/`to_h` given it's used elsewhere and is a little hard to understand as-is.